File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left = 0
        right = len(height) - 1

        maxWater = 0
        while (left < right):
            minHeight = min(height[left], height[right])
            water = minHeight * (right - left)
            maxWater = max(water, maxWater)

            if height[left] <= height[right]:
                left += 1
            else: 
                right -= 1
        
        return maxWater

File: 11-container-with-most-water/container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    let maxVolume = 0;
    let [left, right] = [0, height.length - 1];

    while (left < right) {
        const leftHeight = height[left];
        const rightHeight = height[right];
        const currVolume = (right - left) * Math.min(leftHeight, rightHeight);
        maxVolume = Math.max(maxVolume, currVolume);

        if (leftHeight < rightHeight){
            left++;
        } else {
            right--;
        }
    }
    return maxVolume
};